cmd/xl-storage-disk-id-check.go

```diff
@@ -452,9 +467,16 @@ func (p *xlStorageDiskIDCheck) ReadFileStream(ctx context.Context, volume, path
 	}
 	defer done(&err)
 
-	rc, err := p.storage.ReadFileStream(ctx, volume, path, offset, length)
+	w := xioutil.NewDeadlineWorker(diskMaxTimeout)
+
+	var rc io.ReadCloser
+	err = w.Run(func() error {
+		var ierr error
+		rc, ierr = p.storage.ReadFileStream(ctx, volume, path, offset, length)
```

Comment:
This is a streaming operation. You cannot apply a global deadline to this. It could be streaming gigabytes of data.

Reply:
This doesn't add deadline to the reader @klauspost it is to the return of file descriptor  from this call. 

Since ReadFileStream() `open()` can block this is a deadline worker. Once the `rc` is available this caller simply returns upwards wrapped around deadline reader.